src/app/components/authmodel/AuthModel.js

```diff
@@ -0,0 +1,226 @@
+    </div>
+  )
+}
+
+const RegisterForm = ({setAuthStates, username, setUsername}) => {
+
+    const [password, setPassword] = useState('')
+    const [email, setEmail] = useState('')
+    const [message, setMessage] = useState('')
+
+    const handleRegister = async (e) => {
+        e.preventDefault()
+        setMessage('')
+        try {
+            const { nextStep } = await signUp({
+            username,
+            password,
+            attributes: { email },
+          })
+          console.log(nextStep)
+
+          switch(nextStep.signUpStep){
+            case 'CONFIRM_SIGN_UP':
+              setMessage('注册成功！请检查您的电子邮件以验证账户。')
+              setTimeout(() => {setAuthStates('confirm')}, 800)
+              console.log('CONFIRM_SIGN_UP')
+              break
+              // TODO: Handle MFA if app supports it
+            case 'COMPLETE_AUTO_SIGN_IN':
+              break
+              // TODO: prompt user to confirm sign up 
+            case 'DONE':
+              setMessage('注册完全成功')
+              break
+          default:
+              break
+          }
+
+        } catch (error) {
+          setMessage(error.message)
+        }
+      }
+  return (
+    <div>
+      <h2 className="text-2xl mb-4 text-slate-700">Register</h2>
+      <form onSubmit={handleRegister}>
+        <input
+          type="text"
+          placeholder="Username"
+          className="w-full p-2 mb-4 border dark:text-slate-900 rounded"
+          value={username}
+          onChange={(e) => setUsername(e.target.value)}
+        />
+        <input
+          type="email"
+          placeholder="Email"
+          className="w-full p-2 mb-4 border dark:text-slate-900 rounded"
+          value={email}
+          onChange={(e) => setEmail(e.target.value)}
+        />
+        <input
+          type="password"
+          placeholder="Password"
+          className="w-full p-2 mb-4 border dark:text-slate-900 rounded"
+          value={password}
+          onChange={(e) => setPassword(e.target.value)}
+        />
+        <button className="w-full bg-blue-500 text-white p-2 rounded">Register</button>
+      </form>
+      {message && <p className="mt-4 text-red-500">{message}</p>}
+    </div>
+  )
+}
+
+const ConfirmSignUpForm = ({ setAuthStates, username}) => {
+    const [confirmationCode, setConfirmationCode] = useState('')
+    const [message, setMessage] = useState('')
+
+    useEffect(() => {
+
+        if (!username) {
+          setMessage('无效的用户名。请重新注册。')
+        }
+      },[])
+    
+      const handleConfirmSignUp = async (e) => {
+        e.preventDefault()
+        try {
+            console.log(username+'请求函数')
+            const { isSignUpComplete, nextStep } =  await confirmSignUp({
+                username,
+                confirmationCode
+              })
+            console.log(username+'请求函数结束')
+          console.log(isSignUpComplete, nextStep)
+          setMessage('确认成功！请重新登录...')
```

Comment:
`isSignUpComplete` and `nextStep ` should be checked to make sure it's successfully registered